Codeinwp/neve
# Fix missing translations for Customizer

### Summary


After some long debugging, the missing translations were caused by how we generated the JS build.

Alongside the main script `control.js`, some chunk files were generated by Webpack. Creating the `.pot` file associated the labels with those chunks scripts. 

Thus, in the `wp-content/languages/theme` folder of WP, you will find JSON files like:

```json
{"translation-revision-date":"2023-10-21 15:11:06+0000","generator":"GlotPress\/4.0.0-alpha.11","domain":"messages","locale_data":{"messages":{"":{"domain":"messages","plural-forms":"nplurals=2; plural=n != 1;","lang":"it"},"Search":["Cerca"]}},"comment":{"reference":"assets\/apps\/customizer-controls\/build\/585056780d73ab9a0975.js"}}
```

Notice the `reference` that shows what is the associated file.

The [code here](https://github.com/Codeinwp/neve/blob/06a6ffa8c8bdd03e9e95f9b1ae8ba9ac75856ed8/inc/customizer/loader.php#L161-L163) that loads the translation is incomplete because is loading the translations only for `control.js` file.

The solution was to scan the folder for the chunks and apply the same process as we do for `control.js.`

With this, the translation will be included in page like this:

```html
<script id="neve-customizer-chunk-585056780d73ab9a0975-js-translations">
( function( domain, translations ) {
	var localeData = translations.locale_data[ domain ] || translations.locale_data.messages;
	localeData[""].domain = domain;
	wp.i18n.setLocaleData( localeData, domain );
} )( "neve", {"translation-revision-date":"2023-10-21 15:11:06+0000","generator":"GlotPress\/4.0.0-alpha.11","domain":"messages","locale_data":{"messages":{"":{"domain":"messages","plural-forms":"nplurals=2; plural=n != 1;","lang":"it"},"Search":["Cerca"]}},"comment":{"reference":"assets\/apps\/customizer-controls\/build\/585056780d73ab9a0975.js"}} );
</script>
```

### Will affect visual aspect of the product

YES

### Screenshots 

## Before
![merge_trasl_0](https://github.com/Codeinwp/neve/assets/17597852/08950cb8-8330-44fe-b066-449aab945d1b)

## After
![merg_translatiile](https://github.com/Codeinwp/neve/assets/17597852/1e1942f3-b248-4f4e-90bc-c4b808a3f2c0)



### Test instructions


⚠️ You will need Neve Pro

- Change the website language to something else other than English (make sure to pick one with [100% translation](https://translate.wordpress.org/projects/wp-themes/neve/))
- Check the Customizer Header/Footer builder for missing translations.

## Check before Pull Request is ready:



Closes #4093
